code/game/objects/effects/beam.dm

```diff
@@ -61,10 +69,92 @@
 	var/list/sources = list() // Whoever served in emitting this beam. Used in prisms to prevent infinite loops.
 	var/_re_emit = 1 // Re-Emit from master when deleted? Set to 0 to not re-emit.
 
-/obj/effect/beam/resetVariables()
-	..("sources", "children", args)
+/obj/effect/beam/emitter/eyes
+	icon_state = "emitter_double_1"
+	base_state = "emitter_double_1"
+	base_damage = 1
+	full_damage = 30
+	max_range = 6
+
+/obj/effect/beam/emitter/eyes/mouse
+	icon_state = "emitter_double_mouse"
+	base_state = "emitter_double_mouse"
+	base_damage = 1
+	full_damage = 1
+	max_range = 4
+
+/obj/effect/beam/emitter/eyes/emit(var/spawn_by, var/_range=-1,var/charged = FALSE)
+	..()
+	if (charged)
+		charge_up()
+	else
+		var/turf/T = loc
+		var/old_dir = dir
+		var/old_steps = steps
+		spawn(5)
+			if (src && loc && loc == T && dir == old_dir && steps == old_steps)
```

Comment:
This is a bit of a weird check, some of it is redundant.
`src` will never be false, if you want to know if your thing hasn't been marked for deletion, check `gcDestroyed` instead. `loc == T` will implicitly fail if `loc == null`.